lib/read.js

```diff
@@ -200,16 +205,25 @@ class ReadElastic extends AdapterRead {
         return this._check_options_vs_mapping(indices)
             .then(() => {
                 this.es_opts.indices = indices;
-                this.es_opts.limit = this._calculate_limit(limit);
                 if (!this.fetcher) {
                     this.fetcher = elastic.fetcher(this.id, this.es_filter, from, to, this.es_opts);
                 }
 
-                return this.fetcher();
+                var num_points = this._calculate_query_size(limit);
+
+                if (num_points === 0) {
+                    return {
```

Comment:
Seems like it would be a bit cleaner to check whether points_emitted_so_far === head_or_tail_limit when we're returning the results down below and not force this extra call to read just to determine `eof`.

Then I think this case `(num_points === 0)` should never happen.